backend/src/email/services/email.service.ts

```diff
@@ -61,11 +65,33 @@ const getHydratedMessage = async (
     params
   )
   const body = EmailTemplateService.client.template(template?.body!, params)
+
+  // Get agency details (if exists) from campaign user
+  const campaign = await Campaign.findOne({
+    where: { id: campaignId },
+    include: [
+      {
+        model: User,
+        include: [
+          {
+            model: Domain,
+            include: [Agency],
+          },
+        ],
+      },
+    ],
+  })
+  const agency = campaign?.user?.domain?.agency
```

Comment:
Is there a situation where agency name or agency logo will be `undefined` and the hydrated message will still be considered valid? If not, it would be good to do a check here and throw an error if they are undefined. 

This way you can remove the `undefined` type for `agencyName` and `agencyLogoURI`. Benefits are that callers of this method no longer need to check for `undefined` from the return value and it better encodes the domain.

Reply:
Actually yes, the database will allow for `NULL` values (which I guess will become `undefined` through sequelize?).

For `agencyLogoURI` for example, we're currently planning to leave it blank for agencies who don't have suitable logos (and potentially for the generic government agency too)

For `agencyName`, a useful side effect for leaving it optional as well is that if either `domain` or `agency` is undefined (such as for users in the current DB before backpatching), the new code will run perfectly too